pinot-controller/src/main/java/org/apache/pinot/controller/ControllerStarter.java

```diff
@@ -265,8 +275,9 @@ private void setUpPinotController() {
     _controllerPeriodicTaskScheduler = new ControllerPeriodicTaskScheduler();
     _controllerPeriodicTaskScheduler.init(controllerPeriodicTasks, _controllerLeadershipManager);
 
-    LOGGER.info("Creating rebalance segments factory");
-    RebalanceSegmentStrategyFactory.createInstance(helixParticipantManager);
+    LOGGER.info("Registering rebalance segments factory");
+    _helixResourceManager.registerRebalanceSegmentStrategyFactory(
```

Comment:
This line can be moved into the constructor of RebalanceSegmentStrategyFactory and pass in _helixResourceManager

Reply:
Same as this one.